discovery-infra/oc_utils.py

```diff
@@ -0,0 +1,68 @@
+import requests
+import openshift as oc
+
+from retry import retry
+from logger import log
+
+
+def add_oc_user_arguments(parser):
+    parser.add_argument(
+        '--oc-mode',
+        help='If set, use oc instead of minikube',
+        action='store_true',
+        default=False
+    )
+    parser.add_argument(
+        '-ocu',
+        '--oc-user',
+        help='Username for oc login',
+        type=str
+    )
+    parser.add_argument(
+        '-ocp',
+        '--oc-pass',
+        help='Password for oc login',
+        type=str
+    )
+    parser.add_argument(
+        '-ocs',
+        '--oc-scheme',
+        help='Scheme for assisted-service url on oc',
+        type=str,
+        default='http'
+    )
+
+
+def oc_login(username, password, namespace):
+    log.info('Performing oc login with user: %s', username)
+    oc.login(username, password, cmd_args='--insecure-skip-tls-verify')
+    oc_set_project(namespace)
+
+
+def oc_set_project(project):
+    log.info('Setting default oc project (namespace) to: %s', project)
+    oc.set_default_project(project)
+
+
+def oc_get_service_url(service='assisted-service', scheme='http', retries=5):
+    @retry(tries=retries, delay=3, backoff=2)
+    def geturl():
+        for url in oc_iter_urls_by_service(service, scheme):
+            if is_assisted_service_reachable(url):
```

Comment:
Maybe use the same method like in https://github.com/openshift/assisted-service/blob/master/tools/wait_for_assisted_service.py#L39
Should catch `(requests.exceptions.ConnectionError, requests.exceptions.ReadTimeout)`

Reply:
can't see why this is way is preferred

Reply:
i like waiting.wait i think it is more readable than internal function with retry as decorator. It is overkill.

Reply:
hope that it's better now

Reply:
I don't see you are catching specific exceptions or at all

Reply:
I didn't want to catch any exception here